src/braket/pennylane_plugin/ahs_device.py

```diff
@@ -440,15 +439,202 @@ class BraketLocalAquilaDevice(BraketAhsDevice):
+            pattern (Pattern): list containing magnitude of detuning for all atoms in the device
+            time_interval(array[Number, Number]]): The start and end time for the applied pulses
+
+        Returns:
+            ShiftingField: the object representing the local drive for the AnalogueHamiltonianSimulation object
+        """
+        time_points = self._get_sample_times(time_interval)
+        ts_detuning = self._convert_to_time_series(
+            detuning, time_points, scaling_factor=2 * np.pi * 1e6
+        )
+        shift = ShiftingField(magnitude=Field(time_series=ts_detuning, pattern=pattern))
+
+        return shift
+
+    def _validate_pulses(self, pulses):
+        """Validate that all pulses are defined as expected by the device. This validation includes:
+
+        * Verifying that a global drive is present
+        * Verifying that all local pulses have zero amplitude and phase
+        * Verifying that there are no overlapping wires among the local drives
+        * Verifying that all local detunings are of the same type (float or callable)
+
+        Args:
+            pulses (List[RydbergPulse]): List containing all pulses
+
+        Raises:
+            ValueError: if pulses are invalid
+        """
+
+        # Iterate through pulses to find global drive
+        global_index = -1
+        for i, pulse in enumerate(pulses):
+            if set(pulse.wires) == set(self.wires):
+                if global_index != -1:
+                    raise ValueError(
+                        "Cannot execute a ParametrizedEvolution with multiple global drives."
+                    )
+                global_index = i
+            elif not set(pulse.wires).issubset(set(self.wires)):
+                raise ValueError(
+                    f"ParametrizedEvolution contains wires {pulse.wires} which are not a subset "
+                    f"of device wires {self.wires}."
+                )
+
+        # Validate that global drive covers all wires
+        if global_index == -1:
+            raise ValueError(
+                "ParametrizedEvolution doesn't apply a global driving field to all wires."
+            )
+
+        self._global_pulse = global_index
+
+        local_pulses = pulses.copy()
+        local_pulses.pop(global_index)
+
+        if len(local_pulses) == 0:
+            return
+
+        # Validate that local drives don't have amplitude or phase, and that various detunings aren't inconsistent
+        callable_detunings = callable(local_pulses[0].frequency)
+        local_wires = set()
+
+        for pulse in local_pulses:
+            if pulse.amplitude is not None and (
+                callable(pulse.amplitude) or not math.isclose(pulse.amplitude, 0.0)
+            ):
+                raise ValueError(
+                    "Shifting field only allows specification of detuning. Amplitude must be zero."
+                )
+            if callable(pulse.frequency) ^ callable_detunings:
```

Comment:
What does ^ do?

Reply:
It's the logical XOR operator. If the current detuning and `callable_detunings` don't match, this will be true and raise an error.

Reply:
Why this and not `!=` ? Just seems much more familiar